docs/docs/language_reference/schema/index.md

```diff
@@ -116,19 +116,272 @@ type Post implements Webpage {
+edges. We'll initially just focus on properties as these are the simplest case.
+For a user we can imagine the following properties as a minimum:
+
+* `url`: the URL to the users profile page (REQUIRED)
+* `username`: the unique username of the account (REQUIRED)
+* `display_name`: the name we display for the user (OPTIONAL)
+
+With these two properties we can define a type for the schema called `User` which
+looks as follows:
+
+```graphql
+"""
+A user account on Flutter.
+"""
+type User {
+    """
+    The URL of this user's profile page.
+    """
+    url: String!
+
+    """
+    The user's unique username. All users are required to have one.
+    """
+    username: String!
+
+    """
+    The name the user would like to be known by, if any was set.
+
+    If the user has not set a display name, this value will be `null`.
+    """
+    display_name: String
+}
+```
+
+Here we've defined three string properties with doc comments to describe them. We've 
+also made `url` and `username` required fields by adding the `!` suffix to the type. This
+means that the value can't be null. Alternatively, `display_name` is allowed to be
+`null` as it is just a `String`.
+
+We'll now also define our `Post` type, this will be very similar with a url to the post and
+a message representing the post. For now the user who posted it will be omitted and covered
+in the Edges section.
+
+```graphql
+"""
+A message posted by a Flutter user.
+"""
+type Post implements Webpage {
+    """
+    The URL of this post.
+    """
+    url: String!
+
+    """
+    The contents of the posted message.
+    """
+    message: String!
+}
+```
+
 ## Edges
 
-## Entrypoints
+For our two vertices, we might want to query data that refers to other vertices. This is
+what edges are for - relationships between other vertex types. For our user, these edges will
+be the posts the user has created, and a list of the other users they follow. Adding these
+to the vertex we now get the following schema type:
+
+```graphql
+"""
+A user account on Flutter.
+"""
+type User {
+    # own properties
+    """
+    The URL of this user's profile page.
+    """
+    url: String!
+
+    """
+    The user's unique username. All users are required to have one.
+    """
+    username: String!
+
+    """
+    The name the user would like to be known by, if any was set.
+
+    If the user has not set a display name, this value will be `null`.
+    """
+    display_name: String
+    # end of properties
+
+    # edges
+    """
+    The messages posted by this user, if any.
+    """
+    posts: [Post!]
+
+    """
+    The users this user follows, if any.
+    """
+    follows: [User!]
+}
+```
+
+Here we've split properties and edges and used non-doc comments such as `# own properties`
+to signal this. Also, as a user can make multiple posts and follow multiple people the type
+of both edges is a list. Posts is a list of non-null post objects with the type `[Post!]` and
```

Comment:
This might not be the right place and time to mention it, but edges have interesting nullability rules:
- `Post` means "0 or 1 post"
- `Post!` means "exactly 1 post"
- `[Post!]` means "0 or more posts"
- `[Post!]!` means "1 or more posts"

If the outer type is a list, the inner type must be non-nullable. `[Post]` and `[Post]!` aren't allowed because saying "the list might exist but be full of `null` neighboring vertices" doesn't make any sense.